lib/websocket.js

```diff
@@ -989,7 +989,11 @@ function initAsClient(websocket, address, protocols, options) {
     });
   });
 
-  req.end();
+  if (typeof opts.finishRequest === 'function') {
```

Comment:
```suggestion
  if (typeof opts.finishRequest) {
```

Reply:
You obviously meant just `if (opts.finishRequest)` without the `typeof`, but yeah. I wrote the test like this because that's what node's http code usually seemed to do (e.g. for `options.createConnection`), but I personally also don't really see a good reason to not just check for truthy.

Reply:
Yes, the suggestion is mainly for consistency with the codebase.